src/main/java/de/dennisguse/opentracks/fragments/IntervalsFragment.java

```diff
@@ -54,15 +59,31 @@
     protected final SharedPreferences.OnSharedPreferenceChangeListener sharedPreferenceChangeListener = (sharedPreferences, key) -> {
         if (PreferencesUtils.isKey(getContext(), R.string.stats_units_key, key) || PreferencesUtils.isKey(getContext(), R.string.stats_rate_key, key)) {
             metricUnits = PreferencesUtils.isMetricUnits(sharedPreferences, getContext());
-            if (adapter != null) {
-                adapter.notifyDataSetChanged();
+            uploadIntervals();
+            if (spinnerAdapter != null) {
                 spinnerAdapter.notifyDataSetChanged();
             }
         }
     };
 
-    public static Fragment newInstance() {
-        return new IntervalsFragment();
+    /**
+     * Creates an instance of this class.
+     *
+     * @param  fromTopToBottom If true then the intervals are shown from top to bottom (the first interval on top). Otherwise the intervals are shown from bottom to top.
+     * @return IntervalsFragment instance.
+     */
+    public static Fragment newInstance(boolean fromTopToBottom) {
+        Bundle bundle = new Bundle();
+        bundle.putBoolean(FROM_TOP_TO_BOTTOM_KEY, fromTopToBottom);
+        IntervalsFragment intervalsFragment = new IntervalsFragment();
+        intervalsFragment.setArguments(bundle);
+        return intervalsFragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        stackModeListView = getArguments().getBoolean(FROM_TOP_TO_BOTTOM_KEY, true) == true ? IntervalStatisticsAdapter.StackMode.STACK_FROM_TOP : IntervalStatisticsAdapter.StackMode.STACK_FROM_BOTTOM;
```

Comment:
`== true is not required`, right?
And could we pass the StackMode via the bundle directly?